selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java

```diff
@@ -157,9 +159,13 @@ public Workspace createWorkspace(
     EnvironmentDto environment = workspace.getEnvironments().get("replaced_name");
     environment
         .getMachines()
-        .get("dev-machine")
-        .getAttributes()
-        .put("memoryLimitBytes", Long.toString(convertToByte(memory, memoryUnit)));
+        .values()
+        .stream()
+        .filter(m -> m.getInstallers().contains("org.eclipse.che.ws-agent"))
```

Comment:
dev machine should be found using installers OR servers. E.g.
```
.getMachine()
.values()
.stream()
.filter(m -> {
    return InstallerFqn.idInKeyList("org.eclipse.che.ws-agent", m.getInstallers()) || m.getServers().keySet().contains(Constants.SERVER_WS_AGENT_HTTP_REFERENCE);
})
.forEach(m -> m.getAttributes().put("memoryLimitBytes", Long.toString(convertToByte(memory, memoryUnit))));
```
BTW we have pretty the same logic in workspace module. It is in class WsAgentMachineFinderUtil.
So I would suggest to add new method to this class and use it here.

Reply:
Ok